### PR TITLE
Prevent ACEX HC transfer before unloading

### PR DESCRIPTION
**Pull request by ampersand38**
Reinforcement groups ownership transfer can break the unload waypoint, causing the infantry to fail to disembark.

This PR checks for presence of ACEX headless, blacklists the vehicle and infantry groups, and removes the blacklist after they reach their waypoints.

### CODE REVIEW

**Comment by CreepPork** [comment]
Did you test it with and without ACEX?

**Comment by ampersand38** [comment]
Actually, I copied in from the wrong editor window. The setWaypointStatements should have single quotes around acex_headless_blacklist. Should I close this one and re-submit?
````
//Check ACEX Headless Client
if (acex_headless_enabled && {isClass(configFile >> "CfgPatches" >> "acex_headless")}) then
{
    //to ensure unload, blacklist
    _vehicleGroup setVariable ["acex_headless_blacklist", false, true];
    _infantryGroup setVariable ["acex_headless_blacklist", false, true];

    //after unload, unblacklist _vehicleGroup
    _vehicleUnloadWp setWaypointStatements ["true", "(group this) setVariable ['acex_headless_blacklist', false, true];"];
    //after rally, unblacklist _infantryGroup
    _infantryRallyWp setWaypointStatements ["true", "(group this) setVariable ['acex_headless_blacklist', false, true];"];
};

**Comment by ampersand38** [comment]
I don't know how the heck I uploaded such a wrong file.
Here's the unsigned pbo I tested with.

File: @AresModAchillesExpansion/addons/modules_f_ares/Reinforcements/functions/fn_ReinforcementsCreateUnits.sqf [hunk 1]
```diff
@@ -301,6 +301,19 @@ if (_vehicleBehaviour == 0) then
 	_vehicleReturnWp setWaypointStatements ["true", "deleteVehicle (vehicle this); {deleteVehicle _x} foreach thisList;"];
 };
 
+//Check ACEX Headless Client
+if (acex_headless_enabled && {isClass(configFile >> "CfgPatches" >> "acex_headless")}) then
```
Comment (neilzar) [inline @AresModAchillesExpansion/addons/modules_f_ares/Reinforcements/functions/fn_ReinforcementsCreateUnits.sqf]: Should be the other way around, I think `acex_headless_enabled` will return nil if ACE X is not loaded; which leads to an error.
```suggestion
if (isClass (configFile >> "CfgPatches" >> "acex_headless") && {acex_headless_enabled}) then
```

**Comment by CreepPork** [comment]
@ampersand38 
Will the issue be resolved ([pull request comment](https://github.com/ArmaAchilles/Achilles/pull/479#pullrequestreview-229148253)) or can I close the pull request?

**Comment by ampersand38** [comment]
Will investigate further on Friday.

**Comment by ampersand38** [comment]
> Could suffer from a race condition when the waypoint for some reason gets completed before setting the waypoint statement. The current scheduled framework does not make life easy.

Is this possible because `addWaypoint` and `setWaypointStatements` are quite far apart? Would the chance be reduced if they were consecutive commands?

**Comment by CreepPork** [comment]
@oOKexOo ^^

**Comment by ampersand38** [comment]
Tested on dedi server with this pbo.
[modules_f_ares.zip](https://github.com/ArmaAchilles/Achilles/files/3563648/modules_f_ares.zip)

**Comment by oOKexOo** [comment]
> Is this possible because addWaypoint and setWaypointStatements are quite far apart? Would the chance be reduced if they were consecutive commands?

Yes it would, but the cleaner way is to put them in a unscheduled block (_e.g._ `isNil {...}` or `CBA_fnc_directCall`.

**Pull request merged**